server/middleware/profile/index.js

```diff
@@ -0,0 +1,158 @@
+const bcrypt = require('bcryptjs');
+const generateCookie = require('../../utils/generateCookie');
+
+const resObj = (statusCode, message, data = {}) => ({
+  statusCode,
+  message,
+  data,
+});
+
+const createPayload = (body) => {
+  const { email, firstName, lastName, title, hasOnboarded } = body;
+  return {
+    ...(email ? { email: email.toLowerCase() } : {}),
+    ...(firstName ? { firstName: firstName } : {}),
+    ...(lastName ? { lastName: lastName } : {}),
+    ...(title ? { title: title } : {}),
+    ...(hasOnboarded ? { hasOnboarded: hasOnboarded } : {}),
+  };
+};
+
+const updateProfileData = async (req, res, tokenData) => {
+  let data;
+  let statusCode;
+  let message;
+  try {
+    const userPayload = createPayload(req.body);
+    const { USERS_COLLECTION } = process.env;
+    const userCollection = await mongoUtil.fetchCollection(USERS_COLLECTION);
+    const updateProfile = await userCollection.findOneAndUpdate(
+      { email: tokenData.email },
+      { $set: userPayload },
+      { returnOriginal: false }
+    );
+    if (!updateProfile.value) {
+      statusCode = 401;
+      throw Error('User does not exist!');
+    }
+    const { password, ...rest } = updateProfile.value;
+    generateCookie(res, rest);
+    statusCode = OK;
+    message = 'Success';
+    data = rest;
+  } catch (err) {
+    if (err) {
+      statusCode = statusCode || BAD_REQUEST;
+      message = err.message;
+    } else {
+      statusCode = GATEWAY_TIMEOUT;
+      message = 'Gateway timeout';
+    }
+  }
+  return resObj(statusCode, message, data);
+};
+
+const updatePassword = async (req, res, tokenData) => {
+  let data;
+  let statusCode;
+  let message;
+  const saltRounds = 10;
+  const { USERS_COLLECTION, TEST_TIMEOUT } = process.env;
+  try {
+    let { currentPassword, newPassword } = req.body;
+    const userCollection = await mongoUtil.fetchCollection(USERS_COLLECTION);
+    const userData = await userCollection.findOne({ email: tokenData.email });
+    if (!userData) {
+      statusCode = 401;
+      throw Error('User does not exist!');
+    }
+    const isMatchPassword = await bcrypt.compare(
+      currentPassword,
+      userData.password
+    );
+    if (!isMatchPassword) {
+      statusCode = BAD_REQUEST;
+      throw Error('Wrong Password. Try Again');
+    }
+    const hashedPassword = await bcrypt.hashSync(newPassword, saltRounds);
+    if (hashedPassword) {
+      const updatePassword = await userCollection.findOneAndUpdate(
+        { email: req.body.email },
+        { $set: { password: hashedPassword } }
+      );
+      if (!updatePassword || TEST_TIMEOUT) {
+        throw Error('Gateway Timeout');
+      } else {
+        statusCode = OK;
+        message = 'Success';
+        data = {};
+      }
+    }
+  } catch (err) {
+    if (err) {
+      statusCode = statusCode || BAD_REQUEST;
+      message = err.message;
+    } else {
+      statusCode = GATEWAY_TIMEOUT;
+      message = 'Gateway timeout';
+    }
```

Comment:
log error events

Reply:
resolve in commit #https://github.com/femmecubator/femmecubator-api-repo/pull/72/commits/58b6dd83fe0bda2e2c2dc2899c8593f45bc7bcc0